packages/x-date-pickers-pro/src/internals/hooks/useEnrichedRangePickerField.ts

```diff
@@ -145,14 +145,17 @@ const useMultiInputFieldSlotProps = <
     }
 
     const currentFieldRef = rangePosition === 'start' ? startFieldRef : endFieldRef;
-    currentFieldRef.current?.focusField();
+    if (!currentFieldRef.current?.isFieldFocused()) {
```

Comment:
Avoid unnecessary re-focus and force setting the section to the first one when switching from start to end field.